DualView/src/main/java/com/microsoft/device/display/samples/dualview/fragments/RestaurantsFragment.kt

```diff
@@ -0,0 +1,136 @@
+import com.microsoft.device.display.samples.dualview.R
+import com.microsoft.device.display.samples.dualview.model.Restaurant
+import com.microsoft.device.display.samples.dualview.util.TAG
+import com.microsoft.device.display.samples.dualview.view.RestaurantAdapter
+import com.microsoft.device.display.samples.dualview.view.SelectedViewModel
+import com.microsoft.device.display.samples.dualview.view.SelectedViewModel.Companion.NO_ITEM_SELECTED
+import com.microsoft.device.dualscreen.ScreenInfo
+import com.microsoft.device.dualscreen.ScreenInfoListener
+import com.microsoft.device.dualscreen.ScreenManagerProvider
+import kotlinx.android.synthetic.main.fragment_restaurants.*
+
+/**
+ * [Fragment] implementation that contains the restaurant list
+ */
+class RestaurantsFragment : Fragment(), ScreenInfoListener {
+    private var restaurantAdapter: RestaurantAdapter? = null
+    private val selectedViewModel: SelectedViewModel by activityViewModels()
+    private var currentScreenInfo: ScreenInfo? = null
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setHasOptionsMenu(true)
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
+        if (currentScreenInfo?.isDualMode() == false) {
+            menu.clear()
+            inflater.inflate(R.menu.menu_list, menu)
+        }
+        super.onCreateOptionsMenu(menu, inflater)
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        return when (item.itemId) {
+            R.id.action_map -> {
+                markSelectedRestaurant(NO_ITEM_SELECTED)
+                shopMapFragment(R.id.first_container_id)
+                true
+            }
+            else -> super.onOptionsItemSelected(item)
+        }
+    }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        return inflater.inflate(R.layout.fragment_restaurants, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        setupRestaurantList()
+    }
+
+    /**
+     * Setup for the restaurant [RecyclerView]
+     */
+    private fun setupRestaurantList() {
+        restaurantAdapter = RestaurantAdapter(requireContext(), selectedViewModel, ::onItemClick)
+        restaurants_recycler_view.adapter = restaurantAdapter
+        restaurants_recycler_view.setHasFixedSize(true)
+        val divider = DividerItemDecoration(context, DividerItemDecoration.VERTICAL)
+        ResourcesCompat.getDrawable(resources, R.drawable.item_divider, null)?.let {
+            divider.setDrawable(it)
+        }
+        restaurants_recycler_view.addItemDecoration(divider)
+    }
+
+    override fun onScreenInfoChanged(screenInfo: ScreenInfo) {
+        currentScreenInfo = screenInfo
+        requireActivity().invalidateOptionsMenu()
```

Comment:
I think it would be better to go for "activity?.invalidateOptionsMenu()" to prevent the app from crashing, especially in this async method onScreenInfoChanged. What do you say?

Reply:
I have unregistered the screenInfoListener inside onPause method. So, is not possible that the activity to be null in this point.